_src/_assets/js/dashboard-map.js

```diff
@@ -103,8 +126,51 @@
     const hedAndDek = d3
       .select('#state-map')
       .insert('div', 'div#map-time-scrubber')
-    const hed = hedAndDek.append('h3')
-    const dek = hedAndDek.append('p')
+      .attr('id', 'map-dek')
+    const hed = hedAndDek.append('h2')
+    const dek1 = hedAndDek.append('p')
+    const dek2 = hedAndDek.append('p')
+    dek1.html(
+      `<span id="dek-tests"></span> <span class="legend-text total">tests conducted</span>`,
+    )
+    dek2.html(
+      `<span id="dek-positive"></span> <span class="legend-text positive">positive tests</span>`,
+    )
+    function setupDropdown() {
+      const propertyDropdown = dek1
+        .append('select')
+        .attr('id', 'map-property-select')
+        .attr('value', currentField)
+
+      const propertyOptions = [
+        {
+          value: 'positive',
+          name: 'Positive Cases',
+        },
+        {
+          value: 'totalTestResults',
+          name: 'Total Tests',
+        },
+        {
+          value: 'death',
+          name: 'Deaths',
+        },
+      ]
+      // set up property selector
+      propertyDropdown
+        .selectAll('option')
+        .data(propertyOptions)
+        .enter()
+        .append('option')
+        .text(d => d.name)
+        .attr('value', d => d.value)
+      propertyDropdown.on('change', () => {
+        currentField = propertyDropdown.property('value')
```

Comment:
So, this totally works!

But in D3, event handlers work in a sorta idiomatic way. The arguments passed are `(d, i)`. `d` is the data value, so in this case it will be the `object` in `propertyOptions`, so instead of reaching into the DOM for the property value, it could just be `d.value`

Reply:
good to know, I think I was following your method for the start-stop checkbox.